metadata/metadata.c

```diff
@@ -148,16 +160,75 @@ Allocator *create_allocator(VirtualDisk__ErasureCodeProfile ec_profile)
     return result;
 }
 
-Metadata *new_metadata()
+HyperwarpMetadata *new_metadata(const char *backend, VirtualDisk__ErasureCodeProfile ec_profile)
```

Comment:
what do you think of naming the parameter `backend_name` just to be super explicit about what is expected?